### PR TITLE
Adding registry to Altair

### PR DESCRIPTION
**Pull request by mustermeiszer**

### CODE REVIEW

File: runtime/altair/src/lib.rs [hunk 1]
```diff
@@ -1140,6 +1198,24 @@ impl pallet_pool_system::Config for Runtime {
...
+	type CurrencyId = CurrencyId;
+	type Event = Event;
+	type InterestRate = Rate;
+	type MaxSizeMetadata = MaxSizeMetadata;
+	type MaxTokenNameLength = MaxTrancheNameLengthBytes;
+	type MaxTokenSymbolLength = MaxTrancheSymbolLengthBytes;
+	type MaxTranches = MaxTranches;
+	type ModifyPool = pallet_pool_system::Pallet<Self>;
+	type Permission = Permissions;
+	type PoolCreateOrigin = EnsureSigned<AccountId>;
```
Comment (gruberb) [inline runtime/altair/src/lib.rs:1212]: We could use the type alias `PoolCreateOrigin`, we also use this in `PoolSystem`.

**Review by gruberb — CHANGES_REQUESTED** [review]
Do we need to add `pallet_pool_registry` to the benchmarks as well?
https://github.com/centrifuge/centrifuge-chain/pull/1125/files#diff-dca6eccb11225cda4be1ac1e2a66128f7e7d140f567115a0ab702d8b0b9473d3R1642

**Comment by mustermeiszer** [comment]
Good point will do that. Cc: @mikiquantum

**Comment by mustermeiszer** [comment]
I checked again, and the registry does not expose all needed benchmarking yet. So this is not possible without manually adapting the weights afterwards. So won't add the benchmarks. @gruberb could you remove the request for changes.

File: runtime/altair/src/lib.rs (hunk 1 above)
Comment (mustermeiszer) [reply runtime/altair/src/lib.rs:1212]: That is a good catch! We should change this to democracy!

Comment (mustermeiszer) [reply runtime/altair/src/lib.rs:1212]: The `PoolCreateOrigin` though, should be removed from `PoolSystem` ^^

Comment (gruberb) [reply runtime/altair/src/lib.rs]: Can you explain why?

**Comment by mustermeiszer** [comment]
Yes, but the registry does not implement all needed methods. I.e. it does not benchmark the extrinsic currently. See: https://github.com/centrifuge/centrifuge-chain/blob/4a10ae7a7eaff0062e7aca3bcba4b4129e909bc8/pallets/pool-registry/src/benchmarking.rs#L23-L150

This means, if we add it to the benchmarking clause in the runtimes, the next time we run the benchmarks for that runtime, we will get errors and need to manually add the older ones from the pool-systems. So I would like to finish the benchmarks in the registry first, and life we the imperfect benches till then.

File: runtime/altair/src/lib.rs (hunk 1 above)
Comment (mustermeiszer) [reply runtime/altair/src/lib.rs]: We don't use it anywhere, as we create the pool from the registry.

File: runtime/altair/src/lib.rs [hunk 2]
```diff
@@ -413,7 +413,65 @@ impl InstanceFilter<Call> for ProxyType {
 	fn filter(&self, c: &Call) -> bool {
 		match self {
 			ProxyType::Any => true,
-			ProxyType::NonTransfer => !matches!(c, Call::Balances(..)),
+			ProxyType::NonTransfer => {
```
[the remark this answers is not in the document]
Comment (mustermeiszer) [reply runtime/altair/src/lib.rs:416]: The reason for this is, that `Forbidlisting` is way harder to get right, than `Allowlisting`. Assume, we add another pallet that internally does some token transfers (e.g. investing into some pool). Then we will have an unnoticed breaking change. Where as, when we are explicit about what we are allowing, we will always recognize once we "need" to add something. 

Polkadot has the same setup.

**Review by mikiquantum — DISMISSED** [review]
LGTM!

**Review by mikiquantum — APPROVED** [review]

**Pull request merged**